xpartamupp/xpartamupp.py

```diff
@@ -158,15 +156,16 @@ def __init__(self, sjid, password, room, nick):
         self.add_event_handler('muc::%s::got_online' % self.room, self._muc_online)
         self.add_event_handler('muc::%s::got_offline' % self.room, self._muc_offline)
         self.add_event_handler('groupchat_message', self._muc_message)
+        self.add_event_handler('disconnected', lambda _: self.connect())
```

Comment:
Good catch. As discussed already I'm not satisfied with the re-connect behavior in general, so I'll have another look at that.

Reply:
After some debugging I came to the conclusion that calling `self.connect()` after the client got initialized without any arguments is perfectly fine, as Slixmpp is supposed to reuse the parameters provided for the original call of `ClientXMPP.connect()`. However, there is a bug in Slixmpp preventing this to work for some parameters, most notably for the `disable_starttls` parameter you stumbled upon. I already opened a merge request for Slixmpp with a fix (https://lab.louiz.org/poezio/slixmpp/-/merge_requests/226), but this merge request still needs to get accepted into Slixmpp. For the time being I added a workaround so we aren't affected by this bug and also don't need to pass connection parameters explicitly into the XMPP client object.

I also did fix the other points which bothered me regarding the reconnect behavior and it should behave fine now. That means there won't be any re-connection attempts anymore if a connection fails because of invalid credentials and all other cases will use the same back-off logic as the connect functionality of Slixmpp uses internally when certain errors happen during connection attempts.